eventsources/sources/kafka/start.go

```diff
@@ -375,9 +380,20 @@ func (consumer *Consumer) processOne(session sarama.ConsumerGroupSession, messag
 		return errors.Wrap(err, "failed to marshal the event data, rejecting the event...")
 	}
 
-	if err = consumer.dispatch(eventBody); err != nil {
+	kafkaID := genUniqueID(consumer.eventSourceName, consumer.eventName, consumer.kafkaEventSource.URL, message.Topic, message.Partition, message.Offset)
+
+	if err = consumer.dispatch(eventBody, eventsourcecommon.SetCustomID(kafkaID)); err != nil {
 		return errors.Wrap(err, "failed to dispatch a kafka event...")
 	}
 	session.MarkMessage(message, "")
 	return nil
 }
+
+// Function can be passed as Option to generate unique id for kafka event
+// eventSourceName:eventName:kafka-url:topic:partition:offset
+func genUniqueID(eventSourceName string, eventName string, kafkaURL string, topic string, partition int32, offset int64) string {
+
+	kafkaID := fmt.Sprintf("%s:%s:%s:%s:%d:%d", eventSourceName, eventName, kafkaURL, topic, partition, offset)
```

Comment:
`kafkaURL` could be very long, which contains several URLs separated by comma. I would suggest to split the URL to only take the first one.